node/client/src/lib.rs

```diff
@@ -567,3 +569,250 @@ impl sp_blockchain::HeaderBackend<Block> for Client {
+					frame_system::CheckNonce::<runtime::Runtime>::from($nonce),
+					frame_system::CheckWeight::<runtime::Runtime>::new(),
+					pallet_transaction_payment::ChargeTransactionPayment::<runtime::Runtime>::from($tip),
+				);
+
+				let $raw_payload = runtime::SignedPayload::from_raw(
+					$call.clone(),
+					$extra.clone(),
+					(
+						(),
+						runtime::VERSION.spec_version,
+						runtime::VERSION.transaction_version,
+						$genesis.clone(),
+						$genesis,
+						(),
+						(),
+						(),
+					),
+				);
+
+				$( $usage )*
+			},
+			#[cfg(feature = "rococo")]
+			Self::Rococo($client) => {
+				use rococo_runtime as runtime;
+
+				$( $setup )*
+
+				let $extra: runtime::SignedExtra = (
+					frame_system::CheckNonZeroSender::<runtime::Runtime>::new(),
+					frame_system::CheckSpecVersion::<runtime::Runtime>::new(),
+					frame_system::CheckTxVersion::<runtime::Runtime>::new(),
+					frame_system::CheckGenesis::<runtime::Runtime>::new(),
+					frame_system::CheckMortality::<runtime::Runtime>::from(sp_runtime::generic::Era::mortal(
+						$period,
+						$current_block,
+					)),
+					frame_system::CheckNonce::<runtime::Runtime>::from($nonce),
+					frame_system::CheckWeight::<runtime::Runtime>::new(),
+					pallet_transaction_payment::ChargeTransactionPayment::<runtime::Runtime>::from($tip),
+				);
+
+				let $raw_payload = runtime::SignedPayload::from_raw(
+					$call.clone(),
+					$extra.clone(),
+					(
+						(),
+						runtime::VERSION.spec_version,
+						runtime::VERSION.transaction_version,
+						$genesis.clone(),
+						$genesis,
+						(),
+						(),
+						(),
+					),
+				);
+
+				$( $usage )*
+			},
+		}
+	}
+}
+
+impl frame_benchmarking_cli::ExtrinsicBuilder for Client {
+	fn remark(&self, nonce: u32) -> std::result::Result<OpaqueExtrinsic, &'static str> {
+		with_raw_payload! {
+			self,
+			{extra, client, raw_payload},
+			{
+				// First the setup code to init all the variables that are needed
+				// to build the signed extras.
+				use runtime::{Call, SystemCall};
+
+				let call = Call::System(SystemCall::remark { remark: vec![] });
+				let bob = Sr25519Keyring::Bob.pair();
+
+				let period = polkadot_runtime_common::BlockHashCount::get().checked_next_power_of_two().map(|c| c / 2).unwrap_or(2) as u64;
+
+				let current_block = 0;
+				let tip = 0;
+				let genesis = client.usage_info().chain.best_hash;
+			},
+			(period, current_block, nonce, tip, call, genesis),
+			/* The SignedPayload is generated here */
+			{
+				// Use the payload to generate a signature.
+				let signature = raw_payload.using_encoded(|payload| bob.sign(payload));
+
+				let ext = runtime::UncheckedExtrinsic::new_signed(
+					call,
+					sp_runtime::AccountId32::from(bob.public()).into(),
+					polkadot_core_primitives::Signature::Sr25519(signature.clone()),
+					extra,
+				);
+				Ok(ext.into())
+			}
+		}
+	}
+}
+
+/// Generates inherent data for: Polkadot, Kusama, Westend and Rococo.
+pub fn provide_inherent_data(
```

Comment:
You should make it more obvious that the context is for benchmarking. You should not use this for normal block production.

Reply:
Renamed to `benchmark_inherent_data` and added doc.